templates/azure-cluster-identity/azure-cluster-identity.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
 spec:
-  type: ServicePrincipal
+  type: "${CLUSTER_IDENTITY_TYPE:=ServicePrincipal}"
```

Comment:
Out of curiosity, what other values will `CLUSTER_IDENTITY_TYPE` hold in here?
And can we document the steps on using workload identity?

Reply:
> Out of curiosity, what other values will `CLUSTER_IDENTITY_TYPE` hold in here?

The same as anything else you can put in `spec.type` of an AzureClusterIdentity. I think all of the options are described here: https://capz.sigs.k8s.io/topics/identities. And they are definitely all listed here: https://capz.sigs.k8s.io/reference/v1beta1-api#infrastructure.cluster.x-k8s.io/v1beta1.IdentityType

> And can we document the steps on using workload identity?

Updated the steps here to use the new variable: https://capz.sigs.k8s.io/topics/workload-identity